src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/SpecialFileProviders/AppManifestSpecialFileProvider.cs

```diff
@@ -24,9 +24,9 @@ internal class AppManifestSpecialFileProvider : AbstractFindByNameUnderAppDesign
             _properties = properties;
         }
 
-        protected override Task CreateFileAsync(string path)
+        protected override async Task CreateFileCoreAsync(string path)
         {
-            return _templateFileCreationService.CreateFileAsync("ResourceInternal.zip", path);
+            await _templateFileCreationService.CreateFileAsync("ResourceInternal.zip", path);
```

Comment:
Why add the state machine here but not in `AppSettingsSpecialFileProvider.cs` etc?

Reply:
Ah I originally made a change here that required it, and then removed it without removing this change.